cmdline-pull/discovery_doc.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python
```

Comment:
Yes that's my plan.

Well, I intended to commit this first and move to the pypi module later, but I would abandon this PR and try creating the pypi module first.